e2e/tests/upgrades/genesis_test.go

```diff
@@ -0,0 +1,155 @@
+		transferTxResp := s.Transfer(ctx, chainA, chainAWallet, channelA.PortID, channelA.ChannelID, testvalues.DefaultTransferAmount(chainADenom), chainAAddress, chainBAddress, s.GetTimeoutHeight(ctx, chainB), 0, "")
+		s.AssertTxSuccess(transferTxResp)
+	})
+
+	t.Run("tokens are escrowed", func(t *testing.T) {
+		actualBalance, err := s.GetChainANativeBalance(ctx, chainAWallet)
+		s.Require().NoError(err)
+
+		expected := testvalues.StartingTokenAmount - testvalues.IBCTransferAmount
+		s.Require().Equal(expected, actualBalance)
+	})
+
+	t.Run("start relayer", func(t *testing.T) {
+		s.StartRelayer(relayer)
+	})
+
+	t.Run("packets are relayed", func(t *testing.T) {
+		s.AssertPacketRelayed(ctx, chainA, channelA.PortID, channelA.ChannelID, 1)
+
+		actualBalance, err := chainB.GetBalance(ctx, chainBAddress, chainBIBCToken.IBCDenom())
+		s.Require().NoError(err)
+
+		expected := testvalues.IBCTransferAmount
+		s.Require().Equal(expected, actualBalance.Int64())
+	})
+
+	s.Require().NoError(test.WaitForBlocks(ctx, 10, chainA, chainB), "failed to wait for blocks")
+
+	t.Run("Halt chain and export genesis", func(t *testing.T) {
+		s.HaltChainAndExportGenesis(ctx, chainA, relayer, int64(haltHeight))
+	})
+
+	t.Run("native IBC token transfer from chainA to chainB, sender is source of tokens", func(t *testing.T) {
+		transferTxResp := s.Transfer(ctx, chainA, chainAWallet, channelA.PortID, channelA.ChannelID, testvalues.DefaultTransferAmount(chainADenom), chainAAddress, chainBAddress, s.GetTimeoutHeight(ctx, chainB), 0, "")
+		s.AssertTxSuccess(transferTxResp)
+	})
+
+	t.Run("tokens are escrowed", func(t *testing.T) {
+		actualBalance, err := s.GetChainANativeBalance(ctx, chainAWallet)
+		s.Require().NoError(err)
+
+		expected := testvalues.StartingTokenAmount - 2*testvalues.IBCTransferAmount
+		s.Require().Equal(expected, actualBalance)
+	})
+
+	s.Require().NoError(test.WaitForBlocks(ctx, 5, chainA, chainB), "failed to wait for blocks")
+}
+
+func (s *GenesisTestSuite) HaltChainAndExportGenesis(ctx context.Context, chain *cosmos.CosmosChain, relayer ibc.Relayer, haltHeight int64) {
+	timeoutCtx, timeoutCtxCancel := context.WithTimeout(ctx, time.Minute*2)
+	defer timeoutCtxCancel()
+
+	err := test.WaitForBlocks(timeoutCtx, int(haltHeight), chain)
+	s.Require().Error(err, "chain did not halt at halt height")
+
+	err = chain.StopAllNodes(ctx)
+	s.Require().NoError(err, "error stopping node(s)")
+
+	state, err := chain.ExportState(ctx, int64(haltHeight))
+	s.Require().NoError(err)
+
+	str := strings.SplitAfter(state, "server")
+	state = str[1]
+
+	newGenesisJson := strings.ReplaceAll(state, fmt.Sprintf("\"initial_height\":%d", 0), fmt.Sprintf("\"initial_height\":%d", haltHeight+2))
+
+	for _, node := range chain.Nodes() {
+		err := node.OverwriteGenesisFile(ctx, []byte(newGenesisJson))
+		s.Require().NoError(err)
+	}
+	for _, node := range chain.Nodes() {
+		err := node.CopyFile(ctx, "app.toml", "config/app.toml")
```

Comment:
there is no function that we can use which would act on something like `ConfigFileOverrides` for this is there? Copying the file over is not idea but I think it might be the best approach we have for now!

Reply:
What about reading the existing `config/app.toml`, unmarshaling, setting the halt height to 0, marshal, and write the file. Main concern about referencing the `app.toml` here is having to maintain a config file across versions

Reply:
I'm unsure if it is possible to use [this function](https://github.com/strangelove-ventures/interchaintest/blob/main/testutil/toml.go#L46) directly like interchain test does [here](https://github.com/strangelove-ventures/interchaintest/blob/main/chain/cosmos/chain_node.go#L296), but I'm okay adding our own version of that function

Reply:
Yess, The way I copy file here is the easiest way to do it but it may be affected by version upgrade.

Reply:
Great, I was also reading [this function](https://github.com/strangelove-ventures/interchaintest/blob/main/chain/cosmos/chain_node.go#L296) while looking for ideas but it seems to be quite complicated. I tried copying the file first to make sure everything works. Now I will try to add this function and use it.

Reply:
Should we create another PR for it or do it right away?

Reply:
I think we should up in this pr with the function you mentioned above, look great @anhductn2001

Reply:
```diff
diff --git a/e2e/tests/upgrades/genesis_test.go b/e2e/tests/upgrades/genesis_test.go
index 78c4ab0b7..e03a1a19d 100644
--- a/e2e/tests/upgrades/genesis_test.go
+++ b/e2e/tests/upgrades/genesis_test.go
@@ -121,7 +121,7 @@ func (s *GenesisTestSuite) HaltChainAndExportGenesis(ctx context.Context, chain
        str := strings.SplitAfter(state, "server")
        state = str[1]
 
-       appTomlOverrides := make(testsuite.Toml)
+       appTomlOverrides := make(test.Toml)
 
        appTomlOverrides["halt-height"] = 0
 
@@ -131,7 +131,7 @@ func (s *GenesisTestSuite) HaltChainAndExportGenesis(ctx context.Context, chain
        }
 
        for _, node := range chain.Nodes() {
-               err := testsuite.ModifyTomlConfigFile(
+               err := test.ModifyTomlConfigFile(
                        ctx,
                        zap.NewExample(),
                        node.DockerClient,
```

This passes for me locally

Reply:
I decided to push the fix, figured we could always revert the commit

Reply:
Thank you @colin-axner. LGTM! I thought we have to add this function ourselves, but no.